### ISSUE TITLE
Exit code 1 when using a @call_parse CLI with a return 

### ISSUE THREAD

**Issue by pwdemars** [fastai/fastcore#442]
Using fastcore 1.5.10 I have been noticing some of my Github workflows failing with "Process completed with exit code 1" and no other error messages.

After playing around it seems that @call_parse decorated functions which return something cause this behaviour. A minimal example: 

```
@call_parse
def func1():
    x = 1
    return x

@call_parse
def func2():
    x = 1
```

Now, at the command line: 

```
> func1
> echo $?
1

> func2 
> echo $?
0
```

That is, the first function fails while the second works fine. Is this intended? In my experience it can be useful to double up my CLI with general usage functions; currently I will need to use dedicated CLI functions which are not usable elsewhere in my code. 

Thanks a lot in advance!

**Comment by jph00** [comment]
Yes it is intended, but I can see your argument that it's perhaps more annoying that helpful!... I guess if we removed it the alternative would be to use an Exception to create a non-zero exit code?

**Comment by Taytay** [comment]
I think an exception to create non-zero would be great.
I just got bit by this. I say that because when I run the function in a python file, I want it to return something helpful. When I run it as a script, I'm fine for it not to return anything...

**Comment by jph00** [comment]
Fair enough! A PR for this functionality would be welcome. :)
